statsmodels/tsa/vecm/vecm.py

```diff
@@ -0,0 +1,2146 @@
+            beta = np.real(beta)
+        if gamma.dtype == np.complex128 and np.all(np.imag(gamma) == 0):
+            gamma = np.real(gamma)
+
+        self.alpha = alpha
+        self.beta, self.det_coef_coint = np.vsplit(beta, [self.neqs])
+        self.gamma, self.det_coef = np.hsplit(gamma,
+                                              [self.neqs * (self.k_ar - 1)])
+
+        if "ci" in deterministic:
+            self.const_coint = self.det_coef_coint[:1, :]
+        else:
+            self.const_coint = np.zeros(self.r).reshape((1, -1))
+        if "li" in deterministic:
+            start = 1 if "ci" in deterministic else 0
+            self.lin_trend_coint = self.det_coef_coint[start:start+1, :]
+        else:
+            self.lin_trend_coint = np.zeros(self.r).reshape(1, -1)
+        if self.exog_coint is not None:
+            start = ("ci" in deterministic) + ("li" in deterministic)
+            self.exog_coint_coefs = self.det_coef_coint[start:, :]
+        else:
+            self.exog_coint_coefs = None
+
+        split_const_season = 1 if "co" in deterministic else 0
+        split_season_lin = split_const_season + ((seasons-1) if seasons else 0)
+        if "lo" in deterministic:
+            split_lin_exog = split_season_lin + 1
+        else:
+            split_lin_exog = split_season_lin
+        self.const, self.seasonal, self.lin_trend, self.exog_coefs = \
+            np.hsplit(self.det_coef,
+                      [split_const_season, split_season_lin, split_lin_exog])
+
+        self.sigma_u = sigma_u
+
+        if y_min1 is not None and delta_x is not None \
+                and delta_y_1_T is not None:
+            self._delta_y_1_T = delta_y_1_T
+            self._y_min1 = y_min1
+            self._delta_x = delta_x
+        else:
+            _y_1_T, self._delta_y_1_T, self._y_min1, self._delta_x = \
+                _endog_matrices(endog_tot, self.exog, k_ar,
+                                deterministic, seasons)
+        self.nobs = self._y_min1.shape[1]
+
+    @cache_readonly
+    def llf(self):  # Lutkepohl p. 295 (7.2.20)
+        """Compute VECM(k_ar-1) loglikelihood, where k_ar-1 denotes the number
+        of lagged differences.
+        """
+        K = self.neqs
+        T = self.nobs
+        r = self.r
+        s00, _, _, _, _, lambd, _ = _sij(self._delta_x, self._delta_y_1_T,
+                                         self._y_min1)
+        return - K * T * log(2*math.pi) / 2  \
+            - T * (log(np.linalg.det(s00)) + sum(np.log(1-lambd)[:r])) / 2  \
+            - K * T / 2
+
+    @cache_readonly
+    def _cov_sigma(self):
+        sigma_u = self.sigma_u
+        d = duplication_matrix(self.neqs)
+        d_K_plus = np.linalg.pinv(d)
+        # compare p. 93, 297 Lutkepohl (2005)
+        return 2 * chain_dot(d_K_plus, np.kron(sigma_u, sigma_u), d_K_plus.T)
+
+
+    @cache_readonly
+    def cov_params(self):  # p.296 (7.2.21)
```

Comment:
I guess this will create a conflict with the inherited cov_params which is a method that also computes linear combinations.
maybe it's possible and enough to rename to cov_params_default.